add support for import md files inside md files

mosheboker1 opened the pull request.

PabloSzx commented on .gitignore:
```diff
@@ -5,3 +5,5 @@ dist
 .env
 lib-types
+/.idea/
+/yarn.lock
```
```suggestion
```

this project uses pnpm, not yarn

PabloSzx reviewed (changes requested):
https://github.com/dotansimha/remark-import-partial should be updated to have type definitions, another option is to simply inline the plugin in here and not add it as dependency, it's quite small https://github.com/dotansimha/remark-import-partial/blob/master/index.js

PabloSzx commented on packages/server/types.d.ts:
```diff
@@ -1,2 +1,3 @@
 declare module 'remark-admonitions';
 declare module 'remark-slug';
+declare module 'remark-import-partial';
```
I think it's better to just inline this plugin in @guild-docs/server instead

mosheboker1 commented:
> > > @mosheboker1 Please don't do merges to update this branch, you have to rebase instead of merging, I just cleaned and fixed it
> 
> > 
> 
> > no merging yet, i still need to verify things are working
> 
> 
> 
> I'm not talking about merging this branch into main, this branch had the main merged into it, that's not how it's supposed to work, you have to rebase this branch onto the main, I cherry-picked and fixed this branch, you will have to reset your local branch

Ok

PabloSzx commented:
you have to reset to the commit bc2b and force push

PabloSzx commented:
> > ![image](https://user-images.githubusercontent.com/8672915/137920090-4bec30b3-e63c-4274-bee9-ef609f9e7d32.png)
> > Again, just happened, what are these commits doing? just as I left the branch you can start adding changes, but you don't have to merge or rebase anymore
> 
> I'm running git rebase main on import-md branch and then i'm resolving the conflicts

you don't need to rebase anymore, I already fixed this branch, I'm going to force-push again, the only thing you have to do is `git reset --hard origin/import-md`, the suggestion I made is so that you can learn about how you should proceed in the future
